File: ai-novel-platform/server/app/schemas/model_config.py
from pydantic import BaseModel
from typing import Optional


class ModelConfigCreate(BaseModel):
    name: str
    base_url: str
    api_key: str
    model_name: str
    scenes: list[str] = []


class ModelConfigUpdate(BaseModel):
    name: Optional[str] = None
    base_url: Optional[str] = None
    api_key: Optional[str] = None
    model_name: Optional[str] = None
    scenes: Optional[list[str]] = None
    is_active: Optional[bool] = None


class ModelConfigResponse(BaseModel):
    id: str
    name: str
    provider: str
    base_url: str
    model_name: str
    scenes: list[str]
    is_active: bool
    sort_order: int

    model_config = {"from_attributes": True}

File: ai-novel-platform/client/src/lib/api/model-config.ts
import { api } from './client';
import type { ModelConfig } from '@/types';

export interface ModelConfigCreate {
  name: string;
  base_url: string;
  api_key: string;
  model_name: string;
  scenes?: string[];
}

export const modelConfigApi = {
  list: () => api.get<ModelConfig[]>('/user/model-configs'),
  create: (data: ModelConfigCreate) => api.post<ModelConfig>('/user/model-configs', data),
  update: (id: string, data: Partial<ModelConfig>) => api.put<ModelConfig>(`/user/model-configs/${id}`, data),
  delete: (id: string) => api.delete<void>(`/user/model-configs/${id}`),
};